### PR TITLE
[Merged by Bors] - feat(NumberTheory/SiegelsLemma): replace `Fin m` and `Fin n` with `Fintype α` and `Fintype β`

### PR DESCRIPTION
**Pull request by fbarroero**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/NumberTheory/SiegelsLemma.lean [hunk 1]
```diff
@@ -23,34 +23,36 @@ system it determines has a non-zero integer solution `t` with
...
-/-We set ‖⬝‖ to be Matrix.seminormedAddCommGroup  -/
+/- We set ‖⬝‖ to be Matrix.seminormedAddCommGroup  -/
 attribute [local instance] Matrix.seminormedAddCommGroup
 
 open Matrix Finset
 
 namespace Int.Matrix
 
-variable (m n : ℕ) (A : Matrix (Fin m) (Fin n) ℤ) (v : Fin n → ℤ) (hn : m < n) (hm : 0 < m)
+variable {α β : Type _} [Fintype α] [Fintype β] [DecidableEq β] [DecidableEq α] (m n : ℕ)
```
Comment (riccardobrasca) [inline Mathlib/NumberTheory/SiegelsLemma.lean:41]: Do you really need `m` and `n`? I mean, you can just have `Fintype.card ... < Fintype.card` and friends. The line about notations will be longer, but this is the only drawback I see.

Comment (fbarroero) [reply Mathlib/NumberTheory/SiegelsLemma.lean]: Good point! I got rid of them as variables and introduced as notation. This even simplified some lines later.
Thanks!

**Comment by riccardobrasca** [comment]
This is strictly more general than the current version, thanks!

bors d+

**Comment by fbarroero** [comment]
> This is strictly more general than the current version, thanks!
> 
> bors d+

Thanks!

bors r+

**Comment by fbarroero** [comment]
> ✌️ fbarroero can now approve this pull request. To approve and merge a pull request, simply reply with `bors r+`. More detailed instructions are available [here](https://bors.tech/documentation/getting-started/#reviewing-pull-requests).

bors r+

**Pull request closed without merging**